File: RedSheet.DbEntities/Models/NanoScopeToNegotiate.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("NanoScopeToNegotiates", Schema = "dbo")]
	[Serializable]
	public partial class NanoScopeToNegotiate  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NanoScopeToNegotiateId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string NanoScopeToNegotiateName { get; set; } 
	[InverseProperty("NanoScopeToNegotiate")]
	public ICollection<NanoScopeToNegotiateObjective> NanoScopeToNegotiateObjectives { get; set; } 


	public NanoScopeToNegotiate()
	{
	this.NanoScopeToNegotiateObjectives = new HashSet<NanoScopeToNegotiateObjective>();

    }

    }
}

File: redsheet-portal/src/app/database-models/nano-scope-to-negotiate.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { NanoScopeToNegotiateObjective,  } from './'
export class NanoScopeToNegotiate {
    constructor(nanoScopeToNegotiate?: NanoScopeToNegotiate )  {
        let properties = [ "className", "nanoScopeToNegotiateId", "nanoScopeToNegotiateName", "nanoScopeToNegotiateObjectives",];
        for (let property of properties)
            if (nanoScopeToNegotiate && nanoScopeToNegotiate[property])
                this[property] = nanoScopeToNegotiate[property];
    }
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	nanoScopeToNegotiateId : number =   0 ;
 
    @required()
    @maxLength(200)
	nanoScopeToNegotiateName : string =   undefined;
	@nested(NanoScopeToNegotiateObjective)
	nanoScopeToNegotiateObjectives: NanoScopeToNegotiateObjective[];



}
